test/unit/core/dom/layout/test-viewport-observer.js

```diff
@@ -145,5 +147,24 @@ describes.sandboxed('DOM - layout - Viewport Observer', {}, (env) => {
       expect(elInObEntries[0].isIntersecting).to.be.false;
       expect(elInObEntries[1].isIntersecting).to.be.false;
     });
+
+    it('can observe and unobserve an element with multiple callbacks', () => {
```

Comment:
I believe there is a cleaner way of testing this. The "code smell" here is that we require a `onlyVisibleForTesting` helper. In an ideal case, you can fully test functionality using only public API, which I believe we could do here.

```
it('can observe and unobserve an element with multiple callbacks', () => {
  // 1. setup two spies (callbacks) and use them as observers
  // 2. toggleViewport
  // 3. assert that both spies were called
  // 4. unobserve one of them callback
  // 5. toggleViewport
  // 6. assert that one of the spies was called, and that the other wasn't
});
```

Reply:
Makes sense. Updated the test.